.github/workflows/ci.yml

```diff
@@ -0,0 +1,53 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+env:
+  POETRY_VERSION: 1.7.1
+  GHCR_IMAGE_REPOSITORY: ghcr.io/${{ github.repository_owner }}/poetry
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python:
+          - "3.8"
+          - "3.9"
+          - "3.10"
+          - "3.11"
+          - "3.12"
+        variant:
+          - bookworm
+          - slim
+    steps:
+      - uses: actions/checkout@v3
```

Comment:
Let's use the latest release: https://github.com/actions/checkout/releases/tag/v4.1.1